src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,474 @@
+                Assert.True(finished, "Handshake completed in the allotted time");
+
+                // Expected Client property values:
+                Assert.True(client.IsAuthenticated, "client is authenticated");
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.True(client.IsEncrypted, "client is encrypted");
+                Assert.True(client.IsMutuallyAuthenticated, "mutually authentication is true");
+                Assert.False(client.IsServer, "client is not a server");
+                Assert.True(client.IsSigned, "clientStream is signed");
+                Assert.False(client.LeaveInnerStreamOpen, "Inner stream is open");
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("Kerberos", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated, "remote identity of client is authenticated");
+                IdentityValidator.AssertHasName(serverIdentity, target);
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux)]
+        public void NegotiateStream_StreamToStream_KerberosAuthWithoutRealm_Success()
+        {
+            if (!_isKrbAvailable)
+            {
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new NegotiateStream(clientStream))
+            using (var server = new UnixGssFakeNegotiateStream(serverStream))
+            {
+                Assert.False(client.IsAuthenticated);
+
+                Task[] auth = new Task[2];
+                NetworkCredential credential = new NetworkCredential(TestConfiguration.KerberosUser, _fixture.password);
+                auth[0] = client.AuthenticateAsClientAsync(credential, TestConfiguration.HostTarget);
+                auth[1] = server.AuthenticateAsServerAsync();
+
+                bool finished = Task.WaitAll(auth, TestConfiguration.PassingTestTimeoutMilliseconds);
+                Assert.True(finished, "Handshake completed in the allotted time");
+
+                // Expected Client property values:
+                Assert.True(client.IsAuthenticated, "client is authenticated");
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.True(client.IsEncrypted, "client stream is encrypted");
+                Assert.True(client.IsMutuallyAuthenticated, "mutual authentication is true");
+                Assert.False(client.IsServer, "client is not server");
+                Assert.True(client.IsSigned, "client stream is signed");
+                Assert.False(client.LeaveInnerStreamOpen, "inner stream is open");
+
```

Comment:
Personal Preference: Rather than re-interpreting the meanings of the boolean properties, I find it best to just use them as the string.

```C#
Assert.True(client.IsAuthenticated, "client.IsAuthenticated");
...
```

Then it prints something like:
```
  client.IsAuthenticated:
    Expected: true
    Actual: false
      at ...
```